pkg/operator/ceph/object/bucket/util.go

```diff
@@ -0,0 +1,283 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bucket
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/coreos/pkg/capnslog"
+	"github.com/yard-turkey/lib-bucket-provisioner/pkg/apis/objectbucket.io/v1alpha1"
+	"github.com/yard-turkey/lib-bucket-provisioner/pkg/provisioner"
+	"github.com/yard-turkey/lib-bucket-provisioner/pkg/provisioner/api"
+	"k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	v12 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	v13 "github.com/rook/rook/pkg/client/clientset/versioned/typed/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/operator/ceph/object"
+)
+
+var logger = capnslog.NewPackageLogger("github.com/rook/rook", "op-bucket-prov")
+
+const (
+	genUserLen           = 8
+	cephUser             = "cephUser"
+	provisionerName      = "ceph.rook.io/bucket"
+	prefixObjectStoreSvc = "rook-ceph-rgw"
+	accessKeyIdKey       = "accessKeyID"
+	secretSecretKeyKey   = "secretAccessKey"
+)
+
+func NewBucketController(cfg *rest.Config, p *Provisioner) (*provisioner.Provisioner, error) {
+	const allNamespaces = ""
+	logger.Infof("Ceph Bucket Provisioner launched")
+	return provisioner.NewProvisioner(cfg, provisionerName, p, allNamespaces)
+}
+
+// Return the OB struct with minimal fields filled in.
+func (p *Provisioner) composeObjectBucket() *v1alpha1.ObjectBucket {
```

Comment:
You might add these `Provisioner` methods into provisioner.go. Personally i would avoid a util.go for everything except the most generic helpers.

Reply:
done